### PR TITLE
[processor/deltatocumulative]: explicit-bounds histograms

### PR DESCRIPTION
**Pull request by tombrk**
**Description:** <Describe what has changed.>

adds aggregation support for fixed-bucket / explicit bounds histograms

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30705

**Testing:** unit tests added

**Documentation:** none

### CODE REVIEW

**Comment by sh0rez** [comment]
@RichieSams can you please take a look?

@djaglowski because @jpkrohling is out, would you be able to merge once reviewed and approved?

File: processor/deltatocumulativeprocessor/internal/data/add.go [hunk 1]
```diff
@@ -24,9 +25,44 @@ func (dp Number) Add(in Number) Number {
 	return dp
 }
 
-// nolint
 func (dp Histogram) Add(in Histogram) Histogram {
-	panic("todo")
+	// bounds different: no way to merge, so reset observation to new boundaries
+	if !pslice.Equal(dp.ExplicitBounds(), in.ExplicitBounds()) {
+		in.MoveTo(dp.HistogramDataPoint)
```
Comment (RichieSams) [inline processor/deltatocumulativeprocessor/internal/data/add.go:31]: Should we add a metric for how often this occurs?

File: processor/deltatocumulativeprocessor/internal/data/add.go [hunk 2]
```diff
@@ -24,9 +25,44 @@ func (dp Number) Add(in Number) Number {
...
+		return dp
+	}
+
```
Comment (RichieSams) [inline processor/deltatocumulativeprocessor/internal/data/add.go:34]: Do we need to do the `StartTimeUnixNano` checks like we do for Sums?

**Comment by RichieSams** [comment]
Added two comments. Overall LGTM

File: processor/deltatocumulativeprocessor/internal/data/add.go (hunk 2 above)
Comment (sh0rez) [reply processor/deltatocumulativeprocessor/internal/data/add.go:34]: we are! the code I think you talk about is in `delta.Accumulator[T]`, which applies to all datatypes

https://github.com/sh0rez/opentelemetry-collector-contrib/blob/57d7906d430f6ccd1e4eda0493a1fb1c7d8cf590/processor/deltatocumulativeprocessor/internal/delta/delta.go#L38-L45

File: processor/deltatocumulativeprocessor/internal/data/add.go (hunk 1 above)
Comment (sh0rez) [reply processor/deltatocumulativeprocessor/internal/data/add.go:31]: sgtm, but will do later as we don't have a good way of accessing a `Meter` here without refactoring the `telemetry` package

File: processor/deltatocumulativeprocessor/internal/data/add.go (hunk 2 above)
Comment (RichieSams) [reply processor/deltatocumulativeprocessor/internal/data/add.go:34]: Ahhh. that's why I missed it :)

**Review by RichieSams — APPROVED** [review]

**Comment by xocasdashdash** [comment]
👋 👋 

is there any way of testing this PR out? We currently got bitten by the lack of support for normal histograms, moving to exponentials is not a real option right now and this looks like it adds the required support

**Comment by sh0rez** [comment]
@xocasdashdash yes, when building your own collector (https://opentelemetry.io/docs/collector/custom-collector/) you can specify this branch as the version for deltatocumulative (using a replace directive)

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**